src/rules/preferObjectSpreadRule.ts

```diff
@@ -54,6 +54,17 @@ function walk(ctx: Lint.WalkContext<void>) {
             // Object.assign(...someArray) cannot be written as object spread
             !node.arguments.some(isSpreadElement)) {
             if (node.arguments[0].kind === ts.SyntaxKind.ObjectLiteralExpression) {
+
+                /**
+                 * @url https://github.com/palantir/tslint/issues/3117
+                 * Per TS2698, "spread types may only be created from
+                 * object types."
+                 */
+                if (node.arguments[1] !== undefined) {
+                    if (node.arguments[1].kind === ts.SyntaxKind.ThisKeyword) {
```

Comment:
that's not limited to the second argument. We should exit if any parameter to `Object.assign` is `this` similar to the check `!node.arguments.some(isSpreadElement)` above